动态添加uedit的时候，第一次可以正常显示编辑框，之后每次都没有显示

cool-firer opened the pull request:
动态添加uedit的时候，第一次可以正常显示编辑框，之后每次都没有显示.

campaign committed:
成功替换图片后没有触发contentchange

csbde committed:
修复LIST中处理li时因缺少选择器而造成全局范围影响的问题

worry127722 committed:
[test]

campaign committed:
修复li｛clear:both}前边没有加入选择器，可能会影响全局的css问题

campaign committed:
1.5.0

worry127722 committed:
[test]

carsonxu committed:
修复自动保存配置项不起作用问题

carsonxu committed:
Merge remote-tracking branch 'origin/dev-1.4.3' into dev-1.4.3

techird committed:
添加空串判断

worry127722 committed:
Merge pull request #45 from fex-team/dev-1.4.3

q

carsonxu committed:
添加自动上传的配置项

worry127722 committed:
Merge branch 'dev-1.4.3' of https://github.com/worry127722/ueditor into dev-1.4.3

carsonxu committed:
Merge remote-tracking branch 'origin/dev-1.4.4' into dev-1.4.4

yancend committed:
Merge pull request #400 from worry127722/dev-1.4.3

Dev 1.4.3

Commit:
明确两个配置节

Commit:
更新权限说明

Commit:
Update

Commit:
Update README.md

Commit:
Update README.md

carsonxu committed:
让表格背景透明

carsonxu committed:
Merge remote-tracking branch 'origin/dev-1.4.4' into dev-1.4.4

carsonxu committed:
修改自动上传插件

carsonxu committed:
修改自动上传插件

carsonxu committed:
修改自动上传插件

carsonxu committed:
插入图片不提供title属性

carsonxu committed:
修复ff下拖文件变成base64的问题

worry127722 committed:
ad

worry127722 committed:
6

worry127722 committed:
1

worry127722 committed:
1

worry127722 committed:
3

worry127722 committed:
s

worry127722 committed:
q

techird committed:
fix check or create path

campaign committed:
Merge remote-tracking branch 'remotes/origin/dev-1.4.4' into dev-1.5.0

yancend committed:
[test]

yancend committed:
[test]

yancend committed:
[test]迁移用例 到https://github.com/yancend/testDesign/

carsonxu committed:
修复ff下拖文件变成base64的问题

carsonxu committed:
Merge remote-tracking branch 'origin/dev-1.5.0' into dev-1.5.0

carsonxu committed:
设置 enableAutoSave 配置项

carsonxu committed:
设置 enableMessageShow 配置项

carsonxu committed:
添加 beforeinsertfile 和 afterinsertfile 事件

carsonxu committed:
添加 beforeinsertvideo 和 afterinsertvideo 事件

carsonxu committed:
修改toolbar按钮

carsonxu committed:
修改grunt

carsonxu committed:
修改grunt

carsonxu committed:
修改config

carsonxu committed:
修改config

carsonxu committed:
修改config

carsonxu committed:
修改config

carsonxu committed:
修改远程抓取的代码

carsonxu committed:
修改config

carsonxu committed:
修改上传

carsonxu committed:
修改上传

zhoumengkang committed:
Update addCustomizeButton.js

增加一个自定义引用样式例子，原例子里面写的状态反射似乎对新增的按钮似乎无效。
而我写这个自定义引用也只是借用了`blockquote`的状态值。

carsonxu committed:
Merge pull request #810 from zhoumengkang/patch-1

Update addCustomizeButton.js

csbde committed:
BUG修复 td包含p属性时未清理p的width和height值。会造成表格与后台编辑宽度不一致。

csbde committed:
Merge branch 'dev-1.5.0' of git://github.com/fex-team/ueditor into fex-team-dev-1.5.0

Conflicts:
	_parse/list.js

csbde committed:
Update list.js

csbde committed:
Update list.js

csbde committed:
Update list.js

csbde committed:
Update list.js

csbde committed:
Update list.js

csbde committed:
Update list.js

csbde committed:
合并官方的list

csbde committed:
Merge branch 'dev-1.4.3' of https://github.com/csbde/ueditor into dev-1.4.3

csbde committed:
合并官方的list

carsonxu committed:
插入img删除title属性

carsonxu committed:
插入img删除title属性

carsonxu committed:
修改popup样式

campaign committed:
产生数据中的color默认都格式化为#16进制格式

campaign committed:
产生数据中的color默认都格式化为#16进制格式

campaign committed:
产生数据中的color默认都格式化为#16进制格式

campaign committed:
产生数据中的color默认都格式化为#16进制格式

campaign committed:
产生数据中的color默认都格式化为#16进制格式

campaign committed:
过滤掉word冗余标签

campaign committed:
过滤掉word冗余标签

carsonxu committed:
Merge pull request #826 from csbde/dev-1.4.3

BUG修复 td包含p属性时未清理p的width和height值。会造成表格与后台编辑宽度不一致。

campaign committed:
添加了mark标签在dtd中

campaign committed:
扩展的UI组件需要配置toolbars参数才可显示

campaign committed:
扩展的UI组件需要配置toolbars参数才可显示

campaign committed:
更新

carsonxu committed:
优化快捷上传，清除loader的class

campaign committed:
修正search/replace的bug

Commit:
Update changelog.md

campaign committed:
添加分号

campaign committed:
Merge remote-tracking branch 'origin/dev-1.5.0' into dev-1.5.0

HanCong03 committed:
JSP版本修复上传路径错误的bug

carsonxu committed:
Merge remote-tracking branch 'origin/dev-1.4.4' into dev-1.5.0

Conflicts:
	_test/testDesign/Manual regression cases.xmind

HanCong03 committed:
修复windows平台下的路径bug

Commit:
Fixed windows path

Signed-off-by: hancong03 <hancong03@baidu.com>

Commit:
Merge branch 'dev-1.5.0' of github.com:HanCong03/ueditor into dev-1.5.0

carsonxu committed:
支持 tel: mailto: 等链接，添加配置项 allowLinkProtocols

carsonxu committed:
Merge remote-tracking branch 'origin/dev-1.5.0' into dev-1.5.0

carsonxu committed:
支持 tel: mailto: 等链接，添加配置项 allowLinkProtocols

campaign committed:
修改版本号

carsonxu committed:
添加 zeroclipboardready 事件

carsonxu committed:
Merge branch 'dev-1.4.4' of https://github.com/fex-team/ueditor into dev-1.5.0

# Please enter a commit message to explain why this merge is necessary,
# especially if it merges an updated upstream into a topic branch.
#
# Lines starting with '#' will be ignored, and an empty message aborts
# the commit.

carsonxu committed:
添加图片拉伸缩放的配置项

carsonxu committed:
添加 zeroclipboardready 事件

carsonxu committed:
添加 zeroclipboardready 事件

carsonxu committed:
fix copy plugin bug

carsonxu committed:
修复创建编辑器后，快速销毁编辑器后的报错

netpi committed:
Update README.md

add nodejs url

carsonxu committed:
Merge pull request #1210 from netpi/dev-1.5.0

增加 nodejs 支持的链接

carsonxu committed:
add ueditor for nodejs

carsonxu committed:
修复创建编辑器后，快速销毁编辑器后的报错

carsonxu committed:
修复 serverparam 的 bug

carsonxu committed:
修复按钮样式优先级不够大的问题

carsonxu committed:
修复按钮样式优先级不够大的问题

worry127722 committed:
1

worry127722 committed:
修改测试机地址

worry127722 committed:
修改测试机ip

worry127722 committed:
修改测试配置

worry127722 committed:
修改测试配置

worry127722 committed:
修改测试配置

worry127722 committed:
修改配置文件

carsonxu committed:
恢复 _test 代码

carsonxu committed:
修复在编辑区域打不开 chrome 控制台的问题

carsonxu committed:
修改文档

worry127722 committed:
修改测试文件

worry127722 committed:
修改测试配置

campaign committed:
修复paragraph命令

carsonxu committed:
添加Content-Type判断

carsonxu committed:
修复word图片上传的单位错误

carsonxu committed:
去除图片搜索里面编码使用的长字符串

campaign committed:
修正对于大小写的问题

campaign committed:
修改版本为1.5.0

campaign committed:
整理了代码

campaign committed:
修复对嵌套的span标签设置样式时，被嵌套的标签不能正确设置上给定的样式

campaign committed:
更新版本

worry127722 committed:
调试测试

worry127722 committed:
Merge branch 'dev-1.5.0' of https://github.com/fex-team/ueditor into dev-1.5.0

worry127722 committed:
123

worry127722 committed:
d

worry127722 committed:
d

worry127722 committed:
dd

worry127722 committed:
dddd

worry127722 committed:
ddd

worry127722 committed:
sss

worry127722 committed:
s

worry127722 committed:
d

worry127722 committed:
d

worry127722 committed:
d

worry127722 committed:
dd

worry127722 committed:
d

worry127722 committed:
d

worry127722 committed:
d

minikey committed:
Update table.core.js

zhichao-poper committed:
Update Uploader.class.php

更新构造函数注释

risent committed:
simpleupload didn't fire *contentChange* event

simpleupload didn't trigger the *contentChange* event after the image upload success.
When I use ueditor with angular, I listen to the *contentChange* event to binding the data between editor and model.  When user upload an image, and then just save the model, the model only keep the placeholder image not the server response one.

risent committed:
Update simpleupload.js

risent committed:
Update simpleupload.js

worry127722 committed:
更新test

minikey committed:
Update table.core.js

如果插入标题行，将标题行插入到thead

StruggleBird committed:
###优化：
* 修改服务端配置文件名称为ueditor.config
* 修改了ConfigManager：服务端配置文件从classes目录读取，如果不存在则从webapp的ueditor目录读取配置，增强ueditor的安全性，防止配置文件被窥探；
* 修改BinaryUploader、Base64Uploader:允许配置存储路径，支持存储路径配置为绝对路径形式。
	具体做法：controller.jsp中，在request作用域内指定变量rootPath和对应的存储跟路径 ,ueditor会优先查找该配置，如果没有则将网站根目录作为rootPath(即原有配置)。

StruggleBird committed:
###优化：
* 修改服务端配置文件名称为ueditor.config
* 修改了ConfigManager：服务端配置文件从classes目录读取，如果不存在则从webapp的ueditor目录读取配置，增强ueditor的安全性，防止配置文件被窥探；
* 修改BinaryUploader、Base64Uploader:允许配置存储路径，支持存储路径配置为绝对路径形式。
	具体做法：controller.jsp中，在request作用域内指定变量rootPath和对应的存储跟路径 ,ueditor会优先查找该配置，如果没有则将网站根目录作为rootPath(即原有配置)。

Signed-off-by: StruggleBird <mjhx_zt@qq.com>

StruggleBird committed:
###优化：
* 修改服务端配置文件名称为ueditor.config
* 修改了ConfigManager：服务端配置文件从classes目录读取，如果不存在则从webapp的ueditor目录读取配置，增强ueditor的安全性，防止配置文件被窥探；
* 修改BinaryUploader、Base64Uploader:允许配置存储路径，支持存储路径配置为绝对路径形式。
	具体做法：controller.jsp中，在request作用域内指定变量rootPath和对应的存储跟路径 ,ueditor会优先查找该配置，如果没有则将网站根目录作为rootPath(即原有配置)。

Signed-off-by: StruggleBird <mjhx_zt@qq.com>

StruggleBird committed:
Merge

StruggleBird committed:
Merge branch 'dev-1.5.0' of https://github.com/StruggleBird/ueditor into dev-1.5.0

Conflicts:
	jsp/src/com/baidu/ueditor/upload/Base64Uploader.java
	jsp/src/com/baidu/ueditor/upload/BinaryUploader.java
	jsp/src/com/baidu/ueditor/upload/StorageManager.java

StruggleBird committed:
Merge branch 'dev-1.5.0' of https://github.com/StruggleBird/ueditor into dev-1.5.0

worry127722 committed:
[test]安全修复

worry127722 committed:
[test]安全修复

dail committed:
Update dialog.js

fix an issue that when use custom theme, the cssRules will lost effect.

badsummer committed:
处理中文双引号不能正常转义

处理中文双引号转义问题，如： &ldquo;测试&rdquo;  =>  &amp;ldquo;测试&amp;rdquo;

在 unhtml 与 html 函数中添加中文双引号转义字符

birdol committed:
Update Uploader.class.php

强迫症，错别字。

zxeoc committed:
修正项目在服务器上部署路径与线上访问路径不一致时，上传插件不能正确加载配置的问题

rootPath来自于application.getRealPath( "/" )（在controller.jsp中），即来自于服务器本地真实路径，而requestUri来自于request中的访问路径，当两者不一致的时候（如本地路径为localProj，而访问路径为pubProj），会因为拼接出的路经（ConfigManager.originalPath的值）不对而无法加载配置文件，因此应该只取request中的相对路径部分，拼接在本地的绝对路径后面，从而得到配置文件的绝对路径

zxeoc committed:
修正项目在服务器上部署路径与线上访问路径不一致时，上传插件不能正确加载配置的问题

rootPath来自于application.getRealPath( "/" )（在controller.jsp中），即来自于服务器本地真实路径，而requestUri来自于request中的访问路径，当两者不一致的时候（如本地路径为localProj，而访问路径为pubProj），会因为拼接出的路经（ConfigManager.originalPath的值）不对而无法加载配置文件，因此应该只取request中的相对路径部分，拼接在本地的绝对路径后面，从而得到配置文件的绝对路径

Commit:
A bug in plugin contextmenu

Set the value of enableContextMenu first, then get the value, it will
always be true.

So I add a judge to check whether users have set its value, otherwise
its default value is true.

eecjimmy committed:
textarea 名称

taofish committed:
修改说明

Phinome committed:
Merge pull request #2519 from taofish/dev-1.5.0

修改说明

Phinome committed:
Merge pull request #2453 from YuyingWu/dev-1.5.0

A bug in plugin contextmenu

Phinome committed:
Merge pull request #2417 from zxeoc/dev-1.5.0

修正项目在服务器上部署路径与线上访问路径不一致时，上传插件不能正确加载配置的问题

Phinome committed:
Merge pull request #2389 from birdol/patch-1

Update Uploader.class.php

Phinome committed:
Merge pull request #2239 from fex-team/dev-1.4.3

Dev 1.4.3

Phinome committed:
Merge pull request #2503 from eecjimmy/dev-1.5.0

textarea 名称

Phinome committed:
Merge pull request #1869 from StruggleBird/dev-1.5.0

针对jsp语言的配置改进

Phinome committed:
Merge pull request #1659 from risent/patch-1

simpleupload didn't fire *contentChange* event

Phinome committed:
Merge pull request #1657 from zhuzhichao/patch-4

Update Uploader.class.php 更新构造函数错误的注释

Phinome committed:
add plugins guide

51mono committed:
.Net版-多图上传-在线管理：修复服务器返回图片列表不能倒序Bug

Phinome committed:
Merge pull request #2567 from 51mono/51mono-patch-1

.Net版-多图上传-在线管理：修复服务器返回图片列表不能倒序Bug

Phinome committed:
更新第三方插件贡献规范

Phinome committed:
Create ISSUE_TEMPLATE.md

新增提 ISSUE 规则模板

Phinome committed:
Update ISSUE_TEMPLATE.md

简化内容

Phinome committed:
Update ISSUE_TEMPLATE.md

进一步简化内容

Phinome committed:
Update ISSUE_TEMPLATE.md

更新内容

Phinome committed:
Update ISSUE_TEMPLATE.md

优化内容排版

Phinome committed:
Update ISSUE_TEMPLATE.md

优化内容

Phinome committed:
更新下载

更新下载说明

Phinome committed:
Merge pull request #1696 from minikey/patch-2

Update table.core.js

Phinome committed:
Merge pull request #1650 from minikey/patch-1

Update table.core.js

Phinome committed:
更新 js 规范

更新贡献代码 js 规范

Phinome committed:
Merge pull request #2054 from badsummer/dev-1.5.0

处理中文双引号不能正常转义 _src/core/utils.js

Phinome committed:
Merge pull request #1848 from dail/patch-1

Update dialog.js

Phinome committed:
update .file-title height for firefox

update .file-title height for firefox , ISSUE = 2769

sjshare committed:
修改image.js本地图片上传

修改本地图片上传，当上传文件出现超出大小或格式不允许时，移除文件后文件总数及总大小出错问题。

sjshare committed:
修改attachment.js本地文件上传

修改本地文件上传，当上传文件超出大小或格式不允许时，点击移除后文件总数及总大小出错bug。

Dafrok committed:
更改gruntfile和package.json中过时的配置项

Dafrok committed:
Merge pull request #2826 from fex-team/dev-2.0.0

更改gruntfile和package.json中过时的配置项

Dafrok committed:
修改package.json 增加uglify的dependence

tangtanglove committed:
Update Uploader.class.php

iconv函数无法将unicode转换为utf-8（会有警告），更换为mb_convert_encoding函数

Phinome committed:
Update ISSUE_TEMPLATE.md

增加声明

Phinome committed:
add FAQ wiki

Phinome committed:
Merge pull request #2781 from sjshare/dev-1.5.0

修改本地文件上传bug

Phinome committed:
Merge pull request #2869 from tangtanglove/patch-1

Update Uploader.class.php

Phinome committed:
Update README.md

lzxb committed:
Update buttonicon.css

修复在webpack下的bug：
Module not found: Error: Cannot resolve 'file' or 'directory' ../images/charts.png  in

Phinome committed:
Merge pull request #3201 from lzxb/dev-1.5.0

Update buttonicon.css

houyhea committed:
修改htmlParser的正则表达式,增强html解析的容错能力

现有的htmlParser 节点正则表达式无法匹配:<div style="text-align:center;font-family:" font-size:14px;"=""><img src="http://hs-album.oss.aliyuncs.com/static/27/78/35/image/20161206/20161206174331_41105.gif" alt="" /><br /></div>
修改正则表达式:

Phinome committed:
Merge pull request #3205 from houyhea/dev-1.5.0

修改htmlParser的正则表达式,增强html解析的容错能力

Commit:
Fixed autoheight Plugin onscroll error

修复 autoheight 插件在 UE 实例销毁后，因 onscroll 事件监听没有正常取消监听导致 me.window 报 undefined 错误的问题

woleicom committed:
Create image.js

更改图片尺寸大小bug

Phinome committed:
Merge pull request #3219 from woleicom/dev-1.5.0

Create image.js

Phinome committed:
Merge pull request #3216 from jinzhk/patch-1

修复 autoheight 插件在 destroy 后 onscroll 报错的问题

Phinome committed:
prettier format file

Commit:
注释掉关于表头合并的特殊处理逻辑

regrexd committed:
update issue template

regrexd committed:
update comment

Phinome committed:
Merge pull request #3221 from regrex/dev-1.5.0

干掉关于table表头合并逻辑的特殊处理，该逻辑会导致表头合并出错

Commit:
解决执行 grunt defalut 命令时抛出错误 TypeError: Cannot read property '1' of null

Phinome committed:
Merge pull request #3223 from jinhuiWong/dev-1.5.0

解决执行 grunt defalut 命令时抛出错误 TypeError: Cannot read property '1' of null

Commit:
优化远 程抓取图片插件(catchremoteimage) 的相关代码：
1.支持背景图片的抓取
2.抓取图片时添加loading
3.抓取失败时显示失败的图片而不是不处理
4.事件 catchremotesuccess 添加成功抓取和抓取失败的dom列表参数

Phinome committed:
Merge pull request #3224 from jinhuiWong/dev-1.5.0

优化远 程抓取图片插件(catchremoteimage) 的相关代码

Commit:
fix bug
抓取远程图片失败后 _src 应是failIMG , 不然会影响getContent()输出的值

Phinome committed:
Merge pull request #3228 from jinhuiWong/dev-1.5.0

fix bug

wslpeter commented:
UE.delEditor('id');
var ue = UE.getEditor('id')
但IE下delEditor有bug，ueditor.all.js，中me.container.parentNode为null，

wslpeter commented:
貌似我找到完美解决方法了：
http://blog.csdn.net/wslpeter1987/article/details/76530075

Commit:
添加基于内容的判断方式

Commit:
添加基于内容的判断方式

Phinome committed:
新增安全通告

新增安全通告

jianzhiyao committed:
修复当图片带有GET参数时，无法本地化的bug

例如：
http://img.xiumi.us/xmi/ua/cq5i/i/a662444d6c6b9fe35c5ca77666209612-sz_18655.png?x-oss-process=image/resize,w_640/auto-orient,1/crop,x_58,y_0,w_435,h_139

jianzhiyao committed:
使用PHP自带方法分析url，并拼接还原为原本的带GET请求url

Commit:
修复调整单元格对齐方式会导致textarea元素丢失的问题

修复问题：在关闭xss过滤前提下，当选择多个td（td中含有textarea）并修改单元格对齐方式时，td中的textarea会被清空

conwnet committed:
修正大小写问题

小写的 timezone_set Asia/chongqing 在 linux 下有问题。

Fenzland committed:
修复编辑源码状态下，`.focus()`和`.blur()`失效的问题

## 问题
编缉源码状态下，`.blur()` 方法无法使编辑器失去焦点；`.focus()` 方法不但无法使编缉器获得焦点，还会使编辑器无法接收键盘操作。

## 解决
修改 source 插件，使用与 `.getContent()` 和 `.setContent()` 相同的处理方式，在开启源码编辑模式时对编辑器的 `.focus()` 和 `.blur()` 方法进行覆盖，并在关闭源码编辑模式时还原。

Phinome committed:
Merge pull request #3238 from Fenzland/fix-focus&blur-in-source-mode

修复编辑源码状态下，`.focus()`和`.blur()`失效的问题

Phinome committed:
Merge pull request #3237 from conwnet/patch-2

修正大小写问题

Phinome committed:
Merge pull request #3234 from jianzhiyao/patch-2

修复当图片带有GET参数时，无法本地化的bug

Phinome committed:
Merge pull request #3236 from tylanbin/patch-2

修复调整单元格对齐方式会导致td中textarea元素丢失的问题

Phinome committed:
upgrade apache commons-fileupload jar to fixe old versions bug, causing ddos

Phinome committed:
add apache common-fileupload jar pacakge security alarm

lujiaxing committed:
Update image.js

Make the header same with /plugins/autoupload.js
See:
/fex-team/ueditor/blob/089c7dee81cf06bf68a4ff6a362a7dcf4511d78b/_src/plugins/autoupload.js#L119

Phinome committed:
Merge pull request #3247 from lujiaxing/patch-1

Update image.js to make image uploader send same header with drag-drop-uploading

Commit:
.net 后端示例代码有安全风险，暂时删除

wonderzhou committed:
更新自定义插件开发相关文档说明

Phinome committed:
Merge pull request #3250 from wonderzhou/dev-1.5.0

更新二次开发自定义插件相关文档说明

Commit:
后端代码只是 demo，有太多安全漏洞，为了避免误用，将不再提供

Commit:
Update README.md

Commit:
Merge pull request #3260 from nwind/patch-2

Update README.md
